daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -694,41 +700,38 @@ private[lf] final class Compiler(
 
   private[this] val UnitDef = SDefinition(t.SEValue.Unit)
 
-  // Witness the fact that the template 'tmplId' implements the interface 'ifaceId'
-  private[this] def compileImplements(
-      tmplId: Identifier,
-      ifaceId: Identifier,
-  ): (t.SDefinitionRef, SDefinition) =
-    t.ImplementsDefRef(tmplId, ifaceId) -> UnitDef
+  // Compile the contents of an interface instance, including a witness for
+  // the existence of said interface instance.
+  private[this] def compileInterfaceInstance(
```

Comment:
Replaces `compileImplements`+`foreach(compileImplementsMethod)` (resp. `compileCoImplements`+`foreach(compileCoImplementsMethod)`) for `TemplateImplements` (resp. `InterfaceCoImplements`)